src/main/java/ch/njol/skript/expressions/ExprEnchantItemEnchantments.java

```diff
@@ -0,0 +1,137 @@
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.expressions;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.bukkit.enchantments.Enchantment;
+import org.bukkit.event.Event;
+import org.bukkit.event.enchantment.EnchantItemEvent;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.ScriptLoader;
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Events;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.log.ErrorQuality;
+import ch.njol.skript.util.EnchantmentType;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+
+@Name("Enchant Event Applied Enchantments")
+@Description({"The applied enchantments in an enchant event.",
+				"Deleting or removing the applied enchantments will prevent the item's enchantment."})
+@Examples({"on enchant",
+			"\tset the applied enchantments to sharpness 10 and fire aspect 5"})
+@Events("enchant")
+@Since("INSERT VERSION")
+public class ExprEnchantItemEnchantments extends SimpleExpression<EnchantmentType> {
+
+	static {
+		Skript.registerExpression(ExprEnchantItemEnchantments.class, EnchantmentType.class, ExpressionType.SIMPLE, "[the] applied enchant[ment][s]");
+	}
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (!ScriptLoader.isCurrentEvent(EnchantItemEvent.class)) {
+			Skript.error("The applied enchantments are only usable in an enchant event.", ErrorQuality.SEMANTIC_ERROR);
+			return false;
+		}
+		return true;
+	}
+
+	@SuppressWarnings("null")
+	@Override
+	@Nullable
+	protected EnchantmentType[] get(Event e) {
+		return ((EnchantItemEvent) e).getEnchantsToAdd().entrySet().stream()
+				.map(entry -> new EnchantmentType(entry.getKey(), entry.getValue()))
+				.toArray(EnchantmentType[]::new);
+	}
+
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		if (mode == ChangeMode.REMOVE_ALL || mode == ChangeMode.RESET)
+			return null;
+		return CollectionUtils.array(Enchantment[].class, EnchantmentType[].class);
```

Comment:
I believe this should be fine if it takes only EnchantmentType due to the Enchantment -> EnchantmentType converter (and Skript shouldn't really be returning Enchantment anywhere, if it does, it should be changed), not 100% sure though.

Reply:
I based this off of the following:
```
// Enchantment doesn't get automatically converted to EnchantmentType if you give it more than a one.
// Meaning you can transform an Enchantment array to an EnchantmentType array automatically,
// So, we gotta do it manually.
return CollectionUtils.array(Enchantment[].class, EnchantmentType[].class);
```
Source: https://github.com/SkriptLang/Skript/blob/14f267cc4bb8f00fa5c13536ce8a80afe64b1465/src/main/java/ch/njol/skript/expressions/ExprEnchantments.java#L89

Reply:
Oh right, I wrote that lol. Yeah, we should fix that but for now, just keep accepting both. Sorry for the confusion.